src/overlays/actors/ovl_Boss_Mo/z_boss_mo.c

```diff
@@ -22,299 +111,2944 @@ const ActorInit Boss_Mo_InitVars = {
+    }
+    Math_ApproachF(&this->swingSizeX, 0.0f, 0.1f, 0.05f);
+
+    sp70.x = this->cameraDist;
+    sp70.y = 0.0f;
+    sp70.z = 0.0f;
+    Matrix_RotateY(this->cameraYaw, MTXMODE_NEW);
+    Matrix_MultVec3f(&sp70, &sp64);
+    this->cameraEye.x = sp64.x + this->cameraAt.x;
+    this->cameraEye.z = sp64.z + this->cameraAt.z;
+    one = 1; // Super fake, but it works
+    if (this->csCamera != 0) {
+        if (one) {
+            Math_ApproachF(&this->cameraAt.y, this->cameraNextAt.y, this->cameraAtMaxVel.y,
+                           this->cameraAtVel.y * this->cameraSpeedMod);
+            Math_ApproachF(&this->cameraSpeedMod, 1.0f, 1.0f, this->cameraAccel);
+        }
+        Gameplay_CameraSetAtEye(globalCtx, this->csCamera, &this->cameraAt, &this->cameraEye);
+    }
+}
+
+void BossMo_CoreCollisionCheck(BossMo* this, GlobalContext* globalCtx) {
+    s16 i;
+    Player* player = PLAYER;
+
+    osSyncPrintf(VT_FGCOL(YELLOW));
+    osSyncPrintf("Core_Damage_check START\n");
+    if (this->coreCollider.base.atFlags & AT_HIT) {
+        this->coreCollider.base.atFlags &= ~AT_HIT;
+        if (this->actionState == MO_CORE_UNDERWATER) {
+            this->waitUnderwater = true;
+            this->timers[0] = 150;
+        }
+    }
+    if (this->coreCollider.base.acFlags & AC_HIT) {
+        ColliderInfo* hurtbox = this->coreCollider.info.acHitInfo;
+        // hit!!
+        osSyncPrintf("Core_Damage_check 当り！！\n");
+        this->coreCollider.base.acFlags &= ~AC_HIT;
+        if ((hurtbox->toucher.dmgFlags & 0x00020000) && (this->actionState == MO_CORE_ATTACK)) {
+            this->actionState = MO_CORE_RETREAT;
+        }
+        // hit 2 !!
+        osSyncPrintf("Core_Damage_check 当り 2 ！！\n");
+        if ((this->actionState != MO_CORE_UNDERWATER) && (this->invincibilityTimer == 0)) {
+            u8 damage = CollisionCheck_GetSwordDamage(hurtbox->toucher.dmgFlags);
+
+            if ((damage != 0) && (this->actionState < MO_CORE_ATTACK)) {
+                // sword hit !!
+                osSyncPrintf("Core_Damage_check 剣 当り！！\n");
+                this->actionState = MO_CORE_STUNNED;
+                this->timers[0] = 25;
+
+                this->actor.speedXZ = 15.0f;
+
+                this->actor.world.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+                this->damageFlashTimer = 15;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_MOFER_CORE_DAMAGE);
+                this->actor.colChkInfo.health -= damage;
+                this->hitCount++;
+                if ((s8)this->actor.colChkInfo.health <= 0) {
+                    if (((sMorphaTent1->csCamera == 0) && (sMorphaTent2 == NULL)) ||
+                        ((sMorphaTent1->csCamera == 0) && (sMorphaTent2 != NULL) && (sMorphaTent2->csCamera == 0))) {
+                        Enemy_StartFinishingBlow(globalCtx, &this->actor);
+                        Audio_QueueSeqCmd(NA_BGM_STOP | (1 << 0x10));
```

Comment:
```suggestion
                        Audio_QueueSeqCmd(0x100100FF);
```
same with the other `Audio_QueueSeqCmd(NA_BGM_STOP | (1 << 0x10))`